### DISCUSSION TITLE
Enforcing annotations correctness in global scope

### DISCUSSION THREAD

**Discussion by gryznar** [modular/modular#410] [Q&A] [answered]
I am aware that Mojo has to be compatible with Python in which type annotations are optional and could be wrong. But it does not make sense and I think that Mojo could compromise here in global scope to combine two worlds of system programmers and dynamism of Python.

Would it happen? If type hints will be provided in global scope would Mojo compiler ensure that they are correct?

**Comment by lattner** [comment, marked as answer]
Mojo can tell the difference between Python "hints" and Mojo types.  There will be no problem here, we can support the hints for compatibility (by ignoring them) just like Python does.

**Reply by gryznar** [reply]
Thanks for the answer! 
1. So, Mojo types will be enforced by compiler to be correct in global scope?
2. Hmm, maybe Mojo should not strictly ignore Python "hints", but raise a warning if they are improper?
3. Not strongly related, but in fn it is None as default return type? I don't remember, but I saw some consideration about threating missing return type as object. It is still applicable? More useful will be to keep None in fn :)
